src/main/java/org/veupathdb/service/eda/ss/model/reducer/ValuesFileFactory.java

```diff
@@ -25,23 +26,29 @@ public ValuesFileFactory(Path entityRepositoryDir) {
    * @throws IOException
    */
   public <V, T extends VariableWithValues<V>> FilteredValueFile<V, Long> createFromFilter(
-      SingleValueFilter<V, T> filter) throws IOException {
-    /**
-     * TODO Read metadata from files here for Long vs. Integer or String bytes length?
-     */
+      SingleValueFilter<V, T> filter, Study study) throws IOException {
     BinaryConverter<V> serializer = filter.getVariable().getBinaryConverter();
     return new FilteredValueFile<>(
-        constructPath(filter),
+        binaryFilesManager.getVariableFile(study,
+            filter.getEntity(),
+            filter.getVariable(),
+            BinaryFilesManager.Operation.READ),
         filter.getPredicate(),
         new ValueWithIdDeserializer<>(serializer),
         VariableValueIdPair::getIdIndex);
   }
 
-  private Path constructPath(SingleValueFilter<?,?> filter) {
-    // TODO: Study ID instead of StudyAbbrev?
-    return Path.of(entityRepositoryDir.toString(),
-        filter.getEntity().getStudyAbbrev(),
-        filter.getEntity().getId(),
-        filter.getVariable().getId());
+  public <V> FilteredValueFile<V, VariableValueIdPair<?>> createValuesFile(
```

Comment:
This method isn't creating a file.  it is creating an iterator from a file, right?

Reply:
Right, updated the name to reflect